### PR TITLE
Add IPv6 support to the network logger.

### PR DESCRIPTION
**Pull request by ProgVal**

### CODE REVIEW

File: Loggers/network.py [hunk 1]
```diff
@@ -89,7 +94,9 @@ def __init__(self, simplemonitor, port, key=None):
         if key is None or key == "":
             raise util.LoggerConfigurationError("Network logger key is missing")
         Thread.__init__(self)
-        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        # On Linux, if /proc/sys/net/ipv6/bindv6only=0 (the default), then
```
Comment (jamesoff) [inline Loggers/network.py]: My FreeBSD system has `net.inet6.ip6.v6only: 1` as default, which means this behaviour isn't portable:

```
~src % cat test-socket.py
import socket

sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
sock.bind(('', 31337))
while True:
    sock.listen(10)

~src % python3 test-socket.py &
[1] 24118
~src % sockstat -l | grep 31337
james    python3.6  24118 3  tcp6   *:31337               *:*
```

May be prudent (especially as this is an option configurable on individual systems) to make the address family a config option?

**Comment by ProgVal** [comment]
I just realized that `/proc/sys/net/ipv6/bindv6only` and its equivalent are defaults; that can be overriden per-process with `s.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, <boolean>)`.
So I changed my PR to use this instead.

That should fix the compatibility issue, with no need for a new config option.

**Comment by jamesoff** [comment]
@ProgVal any idea on that failed test? It worked ok on my laptop before I pushed!

```
2018-11-25 17:14:47     INFO (simplemonitor) Adding network logger network: Sending monitor results to ::1:1234
...
2018-11-25 17:14:47    ERROR (simplemonitor.logger-network) Failed to send network data: [Errno 99] Cannot assign requested address
```

I tried setting the remote address to `[::1]` but that didn't work on my laptop at all :)

It would be good to validate the code with both v4 and v6 addresses in the config. v6 isn't my forte though.

**Comment by ProgVal** [comment]
I don't understand how we could get that error when connecting rather than binding. Probably some weirdness in Travis' environment.

> It would be good to validate the code with both v4 and v6 addresses in the config. v6 isn't my forte though.

Thanks for the commit. I don't understand how simplemonitor's testing works so I could find how to write tests :/

**Comment by ProgVal** [comment]
I wrote an issue on Travis' bugtracker: https://github.com/travis-ci/travis-ci/issues/10350

**Pull request merged**